### ISSUE TITLE
Grounding DINO Inference 

### ISSUE THREAD

**Issue by gongjizhang** [open-mmlab/mmdetection#10958]
**Describe the bug**
Grounding DINO demo inference failed!

**Reproduction**

python demo/image_demo.py demo/demo.jpg configs/grounding_dino/grounding_dino_swin-t_pretrain_obj365_goldg_cap4m.py --weights groundingdino_swint_ogc_mmdet-822d7e9d.pth --texts 'bench . car .'

**Environment**

sys.platform: linux
Python: 3.7.0 (default, Nov 29 2021, 08:58:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: NVIDIA GeForce RTX 3090
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.1, V11.1.105
GCC: gcc (GCC) 5.2.1 20150902 (Red Hat 5.2.1-2)
PyTorch: 1.9.0+cu111
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

TorchVision: 0.10.0+cu111
OpenCV: 4.6.0
MMEngine: 0.8.4
MMDetection: 3.1.0+

![捕获](https://github.com/open-mmlab/mmdetection/assets/22426561/c5b4a015-eb08-4458-a89e-a4b4a8b490e6)

**Comment by hhaAndroid** [comment]
@gongjizhang Thank you very much for your feedback. There are still some minor improvements to be made in the current pull request. The "miss key" issue you mentioned above will not affect the accuracy, so you can use it with confidence. Additionally, I would like to inform you that we have added support for fine-tuning, and we will merge this pull request next week.  and funetune outperforms the official code by a significant margin in terms of performance.

**Comment by hhaAndroid** [comment]
@gongjizhang Please wait until next week to use it after the completion. Thank you.

**Comment by gongjizhang** [comment]
Thank you for your reply. I tried to run demo to verify the environment configuration, and it kept showing that it was in the process of inference, No result output

**Comment by hhaAndroid** [comment]
@gongjizhang If you want to verify the correctness of the environment, you can choose previous algorithms like Faster R-CNN.

**Comment by gongjizhang** [comment]
Thank you
